### PR TITLE
Rac dropzone/filetrigger docs

### PR DESCRIPTION
**Pull request by yihuiliao**
mostly docs but there are some additional changes to dropzone and filetrigger.

- Text will no longer take the slot `heading` and instead will take `label`
- Removed FileTrigger's wrapper div -> this means no more context, no SlotProps, no DOMProps. The behavior of FileTrigger should still be the same though.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

- check to make sure the example cards look good
- make sure filetrigger/dropzone behaves as it did before

### CODE REVIEW

File: packages/react-aria-components/docs/react-aria-components.mdx [hunk 1]
```diff
@@ -689,3 +689,23 @@ The documentation for each component includes many examples styled using vanilla
 </ExampleCard>
 
 </section>
+
+### Drag and Drop
```
Comment (yihuiliao) [inline packages/react-aria-components/docs/react-aria-components.mdx:693]: What section  (e.g. collections, date and time, forms, etc.) should DropZone and FileTrigger go under in the docs? I figured we might want to create a new section called drag and drop for DropZone but not sure where FileTrigger should go.

File: packages/react-aria-components/docs/DropZone.mdx [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+
+A dropzone consists of a target element for the dropped objects. Users may drop objects via mouse or keyboard. In addition, a `FileTrigger` may be passed as a child to allow access to computer's file system. 
+
+If a visual label is not provided, then an `aria-label` or
+`aria-labelledby` prop must be passed to identify the button to assistive technology.
+
+### Composed Components
+A dropzone can pass `FileTrigger` as a child, which may also be used standalone or reused in other components.
+
+{/* TODO: Add an Example Card of FileTrigger */}
```
Comment (yihuiliao) [inline packages/react-aria-components/docs/DropZone.mdx:124]: There aren't any svg's to go along with DropZone or FileTrigger's example cards so I figured we just leave it out for now.

File: packages/react-aria-components/docs/DropZone.mdx [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+<DropZone>
+  <Text slot="heading">
+    Drop item here
+  </Text>
+</DropZone>
+```
+
+### File Upload
+
+To upload files, pass `FileTrigger` as a child of `DropZone`.
```
Comment (reidbarber) [inline packages/react-aria-components/docs/DropZone.mdx:275]: I would probably remove mentions of upload from the docs, since this is just a client-side thing.

Maybe something like this:

```suggestion
To allow selection of files from the user's device, pass `FileTrigger` as a child of `DropZone`.
```

File: packages/react-aria-components/docs/react-aria-components.mdx [hunk 4]
```diff
@@ -689,3 +689,23 @@ The documentation for each component includes many examples styled using vanilla
...
+
+<section className={styles.cardGroup}>
+
+<ExampleCard
+  url="DropZone.html"
+  title="DropZone"
+  description="A dropzone is an area into which one or multiple objects can be dragged and dropped.">
+  <Meter />
```
Comment (reidbarber) [inline packages/react-aria-components/docs/react-aria-components.mdx:701]: Just a reminder that we should make SVG, or remove this section before merging.

**Review by reidbarber — COMMENTED** [review]
Looks great! Just one small comment

File: packages/react-aria-components/docs/DropZone.mdx [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+
+</details>
+
+## Features
+
+There is no native element to implement a dropzone in HTML that is widely supported. `DropZone` helps achieve accessible dropzone components that can be styled as needed. 
+
+* **Styleable** – Hover, keyboard focus, and the drop target state is provided for easy styling. These styles only apply when interacting with an appropriate input device, unlike CSS pseudo classes. 
+* **Accessible** – Uses a visually hidden `<button>` under the hood to allow interactions like dropping objects with a keyboard. 
+* **Flexible** – Can be used standalone or with `FileTrigger` to allow file uploads.
```
Comment (devongovett) [inline packages/react-aria-components/docs/DropZone.mdx:107]: I would call out the support for various data types too. FileTrigger is technically a separate component, and you can render whatever you want inside a drop zone.

```suggestion
* **Flexible** – Files, directories, and custom data types can be dropped, and the contents of the drop zone can be fully customized.
```

File: packages/react-aria-components/docs/DropZone.mdx [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+
+## Anatomy
+
+A dropzone consists of a target element for the dropped objects. Users may drop objects via mouse, keyboard, or touch. In addition, a `FileTrigger` may be passed as a child to allow users to upload files.
```
Comment (devongovett) [inline packages/react-aria-components/docs/DropZone.mdx:111]:
```suggestion
A drop zone consists of a target element for the dropped objects. Users may drop objects via mouse, keyboard, or touch. DropZone accepts any content as its children, which may change when the user drops content. A [FileTrigger](FileTrigger.html) is commonly paired with a DropZone to allow a user to choose files from their device.
```

File: packages/react-aria-components/docs/DropZone.mdx [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+* **Styleable** – Hover, keyboard focus, and the drop target state is provided for easy styling. These styles only apply when interacting with an appropriate input device, unlike CSS pseudo classes. 
+* **Accessible** – Uses a visually hidden `<button>` under the hood to allow interactions like dropping objects with a keyboard. 
+* **Flexible** – Can be used standalone or with `FileTrigger` to allow file uploads.
+
+## Anatomy
+
+A dropzone consists of a target element for the dropped objects. Users may drop objects via mouse, keyboard, or touch. In addition, a `FileTrigger` may be passed as a child to allow users to upload files. 
+
+If a visual label is not provided, then an `aria-label` or
+`aria-labelledby` prop must be passed to identify the visually hidden button to assistive technology.
```
Comment (devongovett) [inline packages/react-aria-components/docs/DropZone.mdx:114]: We might want to add something about the Text slot here as well since that is also used as the label?

File: packages/react-aria-components/docs/DropZone.mdx [hunk 8]
```diff
@@ -0,0 +1,362 @@
...
+
+### Composed Components
+A dropzone can pass `FileTrigger` as a child, which may also be used standalone or reused in other components.
```
Comment (devongovett) [inline packages/react-aria-components/docs/DropZone.mdx:117]:
```suggestion
A drop zone can include a `FileTrigger` as a child, which may also be used standalone or reused in other components.
```

File: packages/react-aria-components/docs/DropZone.mdx [hunk 9]
```diff
@@ -0,0 +1,362 @@
...
+}
+```
+
+The `className` and `style` props also accept functions which receive states for styling. This lets you dynamically determine the classes or styles to apply, which is useful when using utility CSS libraries like [Tailwind](https://tailwindcss.com/).
+
+```jsx
+<DropZone className={({isDropTarget}) => isDropTarget ? 'bg-gray-700' : 'bg-gray-600'} />
+```
+
+Render props may also be used as children to alter what elements are rendered based on the current state. For example, you could render an extra element when the button is in a pressed state.
```
Comment (devongovett) [inline packages/react-aria-components/docs/DropZone.mdx:157]:
```suggestion
Render props may also be used as children to alter what elements are rendered based on the current state. For example, you could render an extra element when the drop target is in an active state.
```

File: packages/react-aria-components/docs/DropZone.mdx [hunk 10]
```diff
@@ -0,0 +1,362 @@
...
+.draggable.dragging {
+  opacity: 0.5;
+}
+```
+
+</details>
+
+### Heading
+
+The `heading` slot enables the user to reference the text content to define the dropzone's accessible name.
```
Comment (devongovett) [inline packages/react-aria-components/docs/DropZone.mdx:261]: Seeing this here I think we should probably rename the `heading` slot to `label` since it isn't on a `<Heading>` element. We seem to use "label" in a bunch of other components already.

File: packages/react-aria-components/docs/DropZone.mdx [hunk 11]
```diff
@@ -0,0 +1,362 @@
...
+import {Text} from 'react-aria-components';
+
+<DropZone>
+  <Text slot="heading">
+    Drop item here
+  </Text>
+</DropZone>
+```
+
+### File Upload
```
Comment (devongovett) [inline packages/react-aria-components/docs/DropZone.mdx:273]:
```suggestion
### FileTrigger
```

File: packages/react-aria-components/docs/DropZone.mdx [hunk 12]
```diff
@@ -0,0 +1,362 @@
...
+    --focus-ring-color: Highlight;
+  }
+}
+```
+
+</details>
+
+### Visual feedback
+
+A dropzone can provide visual feedback to the user when a drag hovers over the drop target by passing the `getDropOpertation` function. If a drop target only supports data of specific types (e.g. images, videos, text, etc.), then it should implement getDropOperation and return 'cancel' for types that aren't supported. This will prevent visual feedback indicating that the drop target accepts the dragged data when this is not true. [Read more about getDropOperation.](https://react-spectrum.adobe.com/react-aria/useDrop.html#getdropoperation)
```
Comment (devongovett) [inline packages/react-aria-components/docs/DropZone.mdx:349]:
```suggestion
A dropzone displays visual feedback to the user when a drag hovers over the drop target. If a drop target only supports data of specific types (e.g. images, videos, text, etc.), then it should implement the `getDropOperation` prop and return `'cancel'` for types that aren't supported. This will prevent visual feedback indicating that the drop target accepts the dragged data when this is not true. [Read more about getDropOperation.](https://react-spectrum.adobe.com/react-aria/useDrop.html#getdropoperation)
```

I think eventually we can also add the `acceptedDragTypes` prop to make this a bit easier, but that is out of scope for this PR.

File: packages/react-aria-components/docs/DropZone.mdx [hunk 13]
```diff
@@ -0,0 +1,362 @@
...
+
+```tsx example
+<DropZone
+  getDropOperation={(types) =>  types.has('image/png') ? 'copy' : 'cancel'} >
```
Comment (devongovett) [inline packages/react-aria-components/docs/DropZone.mdx:353]:
```suggestion
  getDropOperation={(types) => types.has('image/png') ? 'copy' : 'cancel'}>
```

File: packages/react-aria-components/docs/DropZone.mdx [hunk 14]
```diff
@@ -0,0 +1,362 @@
...
+  Drop files here
+</DropZone>
+```
+
```
Comment (devongovett) [inline packages/react-aria-components/docs/DropZone.mdx:357]: I think we should probably copy more of the content from the `useDrop` page into here at some point. For example, it covers how to work with various data types such as files, directories, and custom types. It also has a bit more description about drop operations, e.g. what they are, the various cursors etc. But let's do that as a separate PR.

File: packages/react-aria-components/docs/FileTrigger.mdx [hunk 15]
```diff
@@ -0,0 +1,208 @@
...
+    --focus-ring-color: Highlight;
+  }
+}
+```
+
+</details>
+
+## Features
+
+Access to a computer’s file system can be created in HTML using an `<input>` with a file `type` attribute. However, file input's on their own are not very useful since they only allow users to select files from their devices. As a result, file inputs are commonly used inside a `<form>` alongside  a `<button>` to submit the form. `FileTrigger` simplifies this by setting up the input element's configuration with reference to a pressable component, such as a button, for a customizable file upload.
```
Comment (devongovett) [inline packages/react-aria-components/docs/FileTrigger.mdx:113]: I think we should focus this on allowing custom styling. FileTrigger is basically just a hidden `<input type="file">` that is triggered by a custom button. Maybe this could be a starting point:

"A file input can be created with an `<input type="file">` element, but this supports very limited styling options. `FileTrigger` works with a pressable child such as a `Button` to create accessible file inputs that can be styled as needed."

File: packages/react-aria-components/docs/FileTrigger.mdx [hunk 16]
```diff
@@ -0,0 +1,208 @@
...
+}
+```
+
+</details>
+
+## Features
+
+Access to a computer’s file system can be created in HTML using an `<input>` with a file `type` attribute. However, file input's on their own are not very useful since they only allow users to select files from their devices. As a result, file inputs are commonly used inside a `<form>` alongside  a `<button>` to submit the form. `FileTrigger` simplifies this by setting up the input element's configuration with reference to a pressable component, such as a button, for a customizable file upload. 
+
+* **Customizable** – Works with any pressable component meaning that it is compatible with pressable React Aria Components and React Spectrum Components.
```
Comment (devongovett) [inline packages/react-aria-components/docs/FileTrigger.mdx:115]:
```suggestion
* **Customizable** – Works with any pressable React Aria or React Spectrum component, and custom components built with [usePress](usePress.html).
```

File: packages/react-aria-components/docs/FileTrigger.mdx [hunk 17]
```diff
@@ -0,0 +1,208 @@
...
+
+## Anatomy
+
+A `FileTrigger` wraps around an input element and a pressable child, usually a button or link, that allows the user to upload files. The input element is not displayed and has no effect on the layout.
```
Comment (devongovett) [inline packages/react-aria-components/docs/FileTrigger.mdx:119]:
```suggestion
A `FileTrigger` wraps around a pressable child such as a button, and includes a visually hidden input element that allows the user to select files from their device.
```

File: packages/react-aria-components/docs/FileTrigger.mdx [hunk 18]
```diff
@@ -0,0 +1,208 @@
...
+
+Access to a computer’s file system can be created in HTML using an `<input>` with a file `type` attribute. However, file input's on their own are not very useful since they only allow users to select files from their devices. As a result, file inputs are commonly used inside a `<form>` alongside  a `<button>` to submit the form. `FileTrigger` simplifies this by setting up the input element's configuration with reference to a pressable component, such as a button, for a customizable file upload. 
+
+* **Customizable** – Works with any pressable component meaning that it is compatible with pressable React Aria Components and React Spectrum Components.
+
+## Anatomy
+
+A `FileTrigger` wraps around an input element and a pressable child, usually a button or link, that allows the user to upload files. The input element is not displayed and has no effect on the layout. 
+
+If a visual label is not provided on the pressable child, then an `aria-label` or `aria-labelledby` prop must be passed to identify the link to assistive technology.
```
Comment (devongovett) [inline packages/react-aria-components/docs/FileTrigger.mdx:121]:
```suggestion
If a visual label is not provided on the pressable child, then an `aria-label` or `aria-labelledby` prop must be passed to identify the file trigger to assistive technology.
```

File: packages/react-aria-components/docs/FileTrigger.mdx [hunk 19]
```diff
@@ -0,0 +1,208 @@
...
+A `FileTrigger` can use the following components, which may also be used standalone or reused in other components.
+
+<ExampleCard
+  url="Button.html"
+  title="Button"
+  description="A button allows a user to perform an action with a mouse, touch, or keyboard.">
+  <Button />
+</ExampleCard>
+
+<ExampleCard
```
Comment (devongovett) [inline packages/react-aria-components/docs/FileTrigger.mdx:133]: I'd like to discourage using links as file triggers. It doesn't semantically make much sense since a link is meant to navigate a user to another page, and this is not doing that. I think we should just focus our examples on buttons for now. If someone wants to make their design visually look like a link, they can style a button to just look like underlined text.

File: packages/react-aria-components/docs/FileTrigger.mdx [hunk 20]
```diff
@@ -0,0 +1,208 @@
...
+
+The states and selectors for each component used in a `FileTrigger` are documented below.
+
+### Button
+
+A [Button](Button.html) can be targeted with the `.react-aria-Button` CSS selector, or by overriding with a custom `className`. It supports the following states:
+
+<StateTable properties={docs.exports.ButtonRenderProps.properties} />
+
+### Link
```
Comment (devongovett) [inline packages/react-aria-components/docs/FileTrigger.mdx:170]: same as above, I think we can remove this section for now

File: packages/react-aria-components/docs/FileTrigger.mdx [hunk 21]
```diff
@@ -0,0 +1,208 @@
...
+<ExampleCard
+  url="Link.html"
+  title="Link"
+  description="A link allows a user to navigate to another page.">
+  <Link />
+</ExampleCard>
+
+## Props
+
+<PropTable component={docs.exports.FileTrigger} links={docs.links} />
```
Comment (devongovett) [inline packages/react-aria-components/docs/FileTrigger.mdx:142]: Looks like the aria labeling props are on here but not used. I think we can remove these from the TypeScript definition since they should just be added to the `<Button>` instead.

We should probably also add the `name` prop here and pass it to the `<input>` element so that the selected file can be submitted as part of a form.

File: packages/react-aria-components/docs/FileTrigger.mdx [hunk 22]
```diff
@@ -0,0 +1,208 @@
...
+
+## Styling
+
+React Aria components can be styled in many ways, including using CSS classes, inline styles, utility classes (e.g. Tailwind), CSS-in-JS (e.g. Styled Components), etc. By default, all components include a builtin `className` attribute which can be targeted using CSS selectors. These follow the `react-aria-ComponentName` naming convention.
+
+```css
+.react-aria-FileTrigger {
```
Comment (devongovett) [inline packages/react-aria-components/docs/FileTrigger.mdx:149]: Now thinking about this, I'm wondering if we should actually remove the extra wrapper div around FileTrigger, since it isn't really needed (you'd usually just style the Button inside). Both DialogTrigger and TooltipTrigger don't have an extra div, so perhaps making FileTrigger match that would be a good idea.

If we decide to do that, then we can also remove `SlotProps` and `DOMProps` from the type definition.

File: packages/react-aria-components/docs/FileTrigger.mdx [hunk 23]
```diff
@@ -0,0 +1,208 @@
...
+import Anatomy from '@react-aria/datepicker/docs/daterangepicker-anatomy.svg';
+import ChevronRight from '@spectrum-icons/workflow/ChevronRight';
+import {Divider} from '@react-spectrum/divider';
+import {ExampleCard} from '@react-spectrum/docs/src/ExampleCard';
+import {Keyboard} from '@react-spectrum/text';
+import Button from '@react-spectrum/docs/pages/assets/component-illustrations/ActionButton.svg';
+import Link from '@react-spectrum/docs/pages/assets/component-illustrations/Link.svg';
+
+---
+category: Drag and drop
```
Comment (devongovett) [inline packages/react-aria-components/docs/FileTrigger.mdx:27]: Topic for discussion with the team on this category

**Review by devongovett — COMMENTED** [review]
Looking good! I had a few wording suggestions, and a few thoughts about the API after seeing everything together in the docs. Let me know if you want to discuss!

File: packages/react-aria-components/docs/DropZone.mdx (hunk 12 above)
Comment (yihuiliao) [reply packages/react-aria-components/docs/DropZone.mdx:349]: oh yeah i believe we have a [follow-up ticket](https://github.com/orgs/adobe/projects/19/views/4?filterQuery=dropzone&pane=issue&itemId=31210918) for that already

File: packages/react-aria-components/docs/DropZone.mdx (hunk 14 above)
Comment (yihuiliao) [reply packages/react-aria-components/docs/DropZone.mdx:357]: cool i'll make a follow-up ticket for that

File: packages/react-aria-components/docs/DropZone.mdx [hunk 24]
```diff
@@ -0,0 +1,397 @@
...
+function Example() {
+  let [dropped, setDropped] = React.useState(false);
+
+  return (
+    <DropZone
+      onDrop={() => {
+        setDropped(true);
+      }}>
+      {dropped ? "You dropped something" : "Drop object here"}
```
Comment (devongovett) [inline packages/react-aria-components/docs/DropZone.mdx:51]: Did you mean to wrap this in `<Text slot="label">`? I see it's imported above.

File: packages/react-aria-components/docs/FileTrigger.mdx [hunk 25]
```diff
@@ -0,0 +1,195 @@
...
+  <Button />
+</ExampleCard>
+
+## Props
+
+<PropTable component={docs.exports.FileTrigger} links={docs.links} />
+
+## Styling
+
+React Aria components can be styled in many ways, including using CSS classes, inline styles, utility classes (e.g. Tailwind), CSS-in-JS (e.g. Styled Components), etc. Unlike other React Aria components, `FileTrigger` does not have a builtin `className` attribute following the `react-aria-ComponentName` naming convention. Instead, it is styled by its pressable children, such as `Button`.
```
Comment (devongovett) [inline packages/react-aria-components/docs/FileTrigger.mdx:153]: Wondering if we should just remove the styling section completely for this one since styling a button is covered on the button page? Or change it to say something like we have for DialogTrigger/TooltipTrigger? https://reactspectrum.blob.core.windows.net/reactspectrum/f167e7f31c3e00033cf52520ce52f3de4d8d5ccf/docs/react-aria/Tooltip.html#tooltiptrigger-1 The first sentence here isn't really relevant, and we could just say that since it doesn't render any elements of its own it doesn't support styling.

**Review by devongovett — COMMENTED** [review]
looks good! 🥳

File: packages/react-aria-components/docs/DropZone.mdx (hunk 24 above)
Comment (yihuiliao) [reply packages/react-aria-components/docs/DropZone.mdx:51]: oops yeah

**Review by reidbarber — APPROVED** [review]
LGTM

**Review by devongovett — APPROVED** [review]
